src/savefile/savefile.py

```diff
@@ -185,14 +185,15 @@ def money(self, val: int):
     def soil_data(self):
         return self._soil_data
 
-    def set_soil_data(self, tiles: pygame.sprite.Group):
+    def set_soil_data(self, *tile_groups: pygame.sprite.Group):
         new_data = {}
-        for tile in tiles:
-            if not tile.hoed:
-                continue
-            if tile.plant is not None:
-                plant_info = PlantInfo(tile.plant.seed_type, tile.plant.age)
-            else:
-                plant_info = None
-            new_data[tile.pos] = TileInfo(tile.watered, tile.pos, plant_info)
+        for tiles in tile_groups:
+            for tile in tiles:
```

Comment:
Similarly, I'd use `itertools.chain` here to avoid this kind of thing, and it allows for faster iteration as well (check the Python docs: https://docs.python.org/3/library/itertools.html#itertools.chain)

Reply:
Thank you 😄
itertools looks very interesting. So far I've only used it once or twice, but I should definitely look into it more